robot/client/client.go

```diff
@@ -1078,3 +1083,54 @@ func (rc *RobotClient) Version(ctx context.Context) (robot.VersionResponse, erro
 
 	return mVersion, nil
 }
+
+func addVersionMetadataToContext(ctx context.Context) context.Context {
+	info, _ := debug.ReadBuildInfo()
+	deps := make(map[string]*debug.Module, len(info.Deps))
+	for _, dep := range info.Deps {
+		deps[dep.Path] = dep
+	}
+	var apiVersion string
+	if dep, ok := deps["go.viam.com/api"]; ok {
+		apiVersion = dep.Version
+	}
+	version := config.Version
```

Comment:
Out of curiosity, did we consider going with a similar approach to other SDKs (creating a new file that has the version info, updating the `release.yml` job to update that version tag, and then grabbing from that directly for version info)? It seems like it would require less fallback logic, we have a consistent template for it in other SDKs, and it could be useful beyond (one could certainly imagine that someone using the Go SDK would want to programmatically access version info).

Reply:
It was an alternative but it seemed like one that we wanted to avoid. There are also places in the RDK that grab the version, and they also avoid doing this.

Reply:
Can you provide links to these places?

Reply:
[In the new `Version` method](https://github.com/viamrobotics/rdk/blob/3500f739989b29c67c307b10db453fef3cc390c5/robot/robot.go#L352-L375)
[In the CLI client](https://github.com/viamrobotics/rdk/blob/3500f739989b29c67c307b10db453fef3cc390c5/cli/client.go#L674)

Reply:
The `Version()` function seems to give us exactly what we want. Setting aside questions of whether the version tag it grabs is correct, is there a reason we aren't just grabbing that version info? It doesn't actually look like it relies on creating a `RobotClient` at all (the `localRobot` method just calls to the free-form `Version` function in `robot.go`).

Reply:
Oh, I didn't even notice I could use the free-form function. I'll try that.